### PR TITLE
Add save data for dot product and fix graphical issue on multi attribute selector

### PR DESCRIPTION
**Pull request by P-bibs**
This PR
* adds save data for the dot product transformation
* fixes a graphical issue where disabled buttons in a multi-attribute selector would still highlight on hover and change the cursor type

### CODE REVIEW

File: src/transformation-components/DotProduct.tsx [hunk 1]
```diff
@@ -28,7 +29,11 @@ export function DotProduct({
     HTMLSelectElement
   >(null, () => setErrMsg(null));
 
-  const [attributes, setAttributes] = useState<string[]>([]);
+  const [attributes, setAttributes] = useState<string[]>(
+    saveData !== undefined ? saveData.attributes : []
+  );
+
+  console.log(attributes);
```
Comment (stardust66) [inline src/transformation-components/DotProduct.tsx:36]: Do we want to leave this in?

**Review by stardust66 — APPROVED** [review]

**Comment by thomascastleman** [comment]
While you're at fixing dot product, could you also fix the `generateSaveData` function of Average? It has save data but returns `{}`.

File: src/transformation-components/DotProduct.tsx (hunk 1 above)
Comment (P-bibs) [reply src/transformation-components/DotProduct.tsx:36]: Fixed

**Review by thomascastleman — APPROVED** [review]
Looks good

**Pull request merged**